analyzer/src/BitOne/PhpMemInfo/Analyzer/TopSearcher.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace BitOne\PhpMemInfo\Analyzer;
+
+class TopSearcher
+{
+    /** @var array */
+    protected $items;
+
+    /**
+     * @param array $items
+     */
+    public function __construct(array $items)
+    {
+        $this->items = $items;
+    }
+
+    /**
+     * Create a list of objects with the largest number of children
+     *
+     * @var int $limit
+     * @return array
+     */
+    public function createTopChildren($limit)
+    {
+        $top = [];
+        for($i = 0; $i < $limit; $i ++) {
+            $top[] = 0;
+        }
+        foreach ($this->items as $key => $item) {
```

Comment:
I would go for something far much simpler in term of algorithm, even if less efficient:
 - create a second array with key as item key and value is the number of children
 - call asort on it
 - call array_slice to get the x top objects with children (using $limit)